mlrun/api/initial_data.py

```diff
@@ -508,33 +539,147 @@ def _rename_marketplace_kind_to_hub(
+        artifact_metadata = {}
+
+        # if it is a legacy artifact, it won't a "metadata" field and the artifact_dict will contain the metadata
+        legacy = is_legacy_artifact(artifact_dict)
+        if not legacy:
+            artifact_metadata = artifact_dict.get("metadata", None)
+
+        # uid - calculate as the hash of the artifact object
+        tag = (
+            artifact_dict.get("tag", "") if legacy else artifact_metadata.get("tag", "")
+        )
+        uid = fill_object_hash(artifact_dict, "uid", tag)
+        new_artifact.uid = uid
+
+        # project - copy as is
+        new_artifact.project = (
+            artifact_metadata.get("project", "")
+            if artifact_metadata
+            else artifact_dict.get("project", "")
+        )
+
+        # key - the artifact's key, without iteration if it is attached to it
+        # TODO: does the key include the iteration when its in the metadata?
+        key = (
+            artifact_dict.get("key", "") if legacy else artifact_metadata.get("key", "")
+        )
+        new_artifact.key = key
+
+        # iteration - the artifact's iteration
+        iteration = (
+            artifact_dict.get("iter", 1) if legacy else artifact_metadata.get("iter", 1)
+        )
+        new_artifact.iteration = int(iteration)
+
+        # kind - doesn't exist in v1, will be set to "artifact" by default
+        new_artifact.kind = artifact_dict.get("kind", mlrun.artifacts.Artifact.kind)
+
+        # producer_id - the current uid value
+        new_artifact.producer_id = artifact.uid
+
+        # updated - the artifact's updated time
+        updated = (
+            artifact_dict.get("updated", datetime.datetime.now())
+            if legacy
+            else artifact_metadata.get("updated", datetime.datetime.now())
+        )
+        new_artifact.updated = updated
+
+        # created - the artifact's created time
+        # since this is a new field, we just take the updated time
+        new_artifact.created = updated
+
+        # full_object - the artifact dict
+        new_artifact.full_object = artifact_dict
+
+        # save the new object to the db
+        new_artifacts.append(new_artifact)
+
+        # save old and new artifact ids to be used later for migrating labels and tags
+        if tag:
+            artifacts_tags_to_migrate.append((artifact, tag))
+        labels = artifact_metadata.get("labels", {})
+        if labels:
+            artifacts_labels_to_migrate.append((artifact, labels))
+
+    # commit the new artifacts to the db
+    db._upsert(db_session, new_artifacts, ignore=True)
+
+    # migrate artifact labels to the new table ("artifact_v2_labels")
+    _migrate_artifact_labels(db, db_session, artifacts_labels_to_migrate)
+
+    # migrate artifact tags to the new table ("artifact_v2_tags")
+    _migrate_artifact_tags(db, db_session, artifacts_tags_to_migrate)
+
+    # truncate the entire artifacts table
+    db.del_artifacts(db_session, commit=False)
```

Comment:
I don't think this is how you want to do this. After all, the `del_artifacts` function will change to work on the new v2 table later on, so this will not produce the correct results once the code changes (and it's likely we'll forget this code exists here once we go on to future PRs).
Best thing (fastest, too) would be to interact directly with the DB and drop the table. Drop table is a very fast and efficient action. But, I would do that only after the commit.